src/ImageSharp/Common/Helpers/SimdUtils.ExtendedIntrinsics.cs

```diff
@@ -160,12 +153,10 @@ internal static void ByteToNormalizedFloat(ReadOnlySpan<byte> source, Span<float
                     Vector<uint> w2 = ConvertToUInt32(f2);
                     Vector<uint> w3 = ConvertToUInt32(f3);
 
-                    Vector<ushort> u0 = Vector.Narrow(w0, w1);
-                    Vector<ushort> u1 = Vector.Narrow(w2, w3);
+                    var u0 = Vector.Narrow(w0, w1);
```

Comment:
I’ve really struggled to get editorconfig to do what I want here (var rules are actually unchanged). This seems to fall under the “type is apparent” rule. There’s actually been a warning sitting in the code there for years, it just was ignored.

Reply:
Why arrays aren’t considered apparent I simply do not know.

Reply:
I don't think it's "apparent" when a non-generic type returns a variable with a generic type of the same name. The lack of the type argument hurts readability, especially with API-s like `Vector.Narrow`. Is there a way to fine-tune this rule? What if we disable it entirely?

Reply:
I'm gonna create an empty project and see if I can get the rules working using the VS editorconfig editor and a fresh file.

Reply:
Here's var turned off everywhere. It's verbose but at least it's consistent.

<img alt="image" src="https://user-images.githubusercontent.com/385879/188247195-d9da822c-6df4-46a6-964a-be12dba7633f.png">

Here's with "when type is apparent"

This one is annoying because the array declaration should be apparent. It's probably the closes to what we want though.

<img alt="image" src="https://user-images.githubusercontent.com/385879/188247462-746f789e-2be1-4f8f-9f93-58ebc923a3e0.png">

Here's "when type is apparent" and "built in types"

This one is annoying because the double and int primitives are not apparent in my opinion since they require the reader to know the syntax declaration rules on the right hand side.

<img alt="image" src="https://user-images.githubusercontent.com/385879/188247390-d413997f-44f7-4660-84cc-45933228a1a3.png">

Which do you prefer?

Reply:
I may be very oldscool, but for me the first one is the lesser evil. I wonder what do others think @SixLabors/core?

This rule/analyzer ([IDE0007?](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0007-ide0008)) really sucks. I wonder if there is a better way to raise awareness about it than posting on https://developercommunity.visualstudio.com/ which is a straight way to oblivion.

Reply:
I agree and with the forcing of target type new we handle known types. I'll implement that.

There's actually 2 issues in Roslyn around this. (One closed as a duplicate)

https://github.com/dotnet/roslyn/issues/23714
https://github.com/dotnet/roslyn/issues/55722

I wish I could do this though. It's really annoying that I can't seem to find a way.
```c#
byte[] b = [10];
```